core/src/commonMain/kotlin/com/sourcepoint/mobile_core/network/Json.kt

```diff
@@ -21,4 +23,12 @@ val jsonWithNulls = Json {
     explicitNulls = true
 }
 
-fun String.encodeToJsonObject() = decodeFromString<JsonObject>(this)
+fun String?.encodeToJsonObject() = this?.let { decodeFromString<JsonObject>(it) }
+
+fun Any?.toJsonPrimitive(): JsonPrimitive = when (this) {
+    null -> JsonNull
+    is Number -> JsonPrimitive(this)
+    is String -> JsonPrimitive(this)
+    is Boolean -> JsonPrimitive(this)
+    else -> JsonPrimitive(this.toString())
+}
```

Comment:
@Nevazhnovu maybe the `when` above can use cascading, like:
```kotlin
fun Any?.toJsonPrimitive(): JsonPrimitive = when (this) {
    null -> JsonNull
    is Number, String, Boolean -> JsonPrimitive(this)
    else -> JsonPrimitive(this.toString())
}
```

Reply:
@andresilveirah Unfortunately, kotlin disagrees with that syntax.
![image](https://github.com/user-attachments/assets/e30c817d-f9a9-430a-8db8-3958a7989f38)